tests/PatternAssertions/TextFeaturedMediaAssert.php

```diff
@@ -74,4 +77,57 @@ protected function assertVideoRatio(string $expected_ratio, string $selector, Cr
     }
   }
 
+  /**
+   * Asserts the link of the pattern.
+   *
+   * @param array $expected_link
+   *   Array with keys: 'label', 'path', 'icon'.
+   *   'icon' can be 'external' or 'corner-arrow'.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertLink(array $expected_link, Crawler $crawler): void {
+    $link_element = $crawler->filter('a.ecl-link.ecl-link--icon.ecl-link--icon-after.ecl-u-mt-m.ecl-u-mt-md-l.ecl-u-type-bold');
+    self::assertEquals($expected_link['path'], $link_element->attr('href'));
+
+    $label_element = $link_element->filter('span.ecl-link__label');
+    self::assertEquals($expected_link['label'], $label_element->text());
+
+    $svg = $link_element->filter('svg.ecl-icon.ecl-icon--s.ecl-icon--primary.ecl-link__icon use');
+    self::assertContains('icons.svg#ui--' . $expected_link['icon'], $svg->attr('xlink:href'));
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getPatternVariant(string $html): string {
+    // Variant can be clearly recognised if image or video and link exist only.
```

Comment:
The variant can be easily extracted by checking the presence and properties of the media and link.
If the link is not present then we return left_simple or right_simple based on the media position.
If the media is not present then left simple or left featured will be returned based on the link style.
If neither media nor image are present  then left_simple will be returned.